expression/builtin_compare_vec.go

```diff
@@ -334,11 +335,67 @@ func (b *builtinNEDurationSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colu
 }
 
 func (b *builtinNullEQIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinNullEQIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf0); err != nil {
+		return err
+	}
+
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	isUnsigned0, isUnsigned1 := mysql.HasUnsignedFlag(b.args[0].GetType().Flag),
+		mysql.HasUnsignedFlag(b.args[1].GetType().Flag)
+
+	args0 := buf0.Int64s()
+	args1 := buf1.Int64s()
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		isNull0 := buf0.IsNull(i)
+		isNull1 := buf1.IsNull(i)
+		switch {
```

Comment:
Although it's hard to implement, it may have better performance if we make for loop inside each of the switch case.
```go
switch {
case xxx:
    for {
    }
case yyy:
    for {
    }
}
```

Reply:
I have a little misunderstanding: this switch statement is in a `for(i:=0;i<n;i++)` loop. If we add a `for (i:=0 ;i<n;i++)` loop to each switch case, the time complexity of this code will increase to O(n ^ 2).

Reply:
Because the `isUnsigned` flag  is for all the rows of the argument, we can judgement it outside the for loop.

```go
switch {
case isUnsigned0 && isUnsigned1:
    for i := 0; i < n; i++ {
    }
case !isUnsigned0 && !isUnsigned1:
    for i := 0; i < n; i++ {
    }
case isUnsigned0 && !isUnsigned1:
    for i := 0; i < n; i++ {
    }
case !isUnsigned0 && isUnsigned1:
    for i := 0; i < n; i++ {
    }
}
```

Reply:
done.